mapper/orm/src/main/java/org/hibernate/search/mapper/orm/massindexing/impl/IdentifierConsumerDocumentProducer.java

```diff
@@ -196,39 +197,85 @@ private void indexAllQueue(SessionImplementor session, List<?> entities) throws
 			return;
 		}
 
-		// TODO use 'SessionWorkExecutor workExecutor = mapping.createSearchManager( session ).createSessionWorkExecutor();' instead
-		PojoWorkPlan workPlan = mapping.createSearchManager( session ).createWorkPlan();
+		// TODO <<monitor>> indexingMonitor.entitiesLoaded( entities.size() );
+		PojoSessionWorkExecutor workExecutor = mapping.createSearchManager( session ).createSessionWorkExecutor();
+		CompletableFuture<?>[] tasks = new CompletableFuture<?>[entities.size()];
```

Comment:
Please call these what they are: futures. A "task" generally provides the code to execute, here these are just futures that give us information about the *result* of a task.